### PR TITLE
[SQSERVICES-1647] Servantify proxy (the swagger part)

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/SQSERVICES-1647

Give proxy service a servant routing table for swagger.  This does not change the actual routing due to blocking issues with testing this on our staging env.

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/extended/src/Servant/API/Extended/RawM.hs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+type ApplicationM m = Request -> (Response -> IO ResponseReceived) -> m ResponseReceived
+
+-- | Variant of 'Raw' that lets you access the underlying monadic context to process the request.
+data RawM deriving (Typeable)
+
+-- | Just pass the request to the underlying application and serve its response.
+--
+-- Example:
+--
+-- > type MyApi = "images" :> Raw
```
Comment (battermann) [inline libs/extended/src/Servant/API/Extended/RawM.hs:26]:
```suggestion
-- > type MyApi = "images" :> RawM
```

File: services/proxy/src/Proxy/API/Public.hs [hunk 2]
```diff
@@ -135,15 +133,23 @@ spotifyToken rq = do
...
         $ Req.empty {Client.secure = True}
 
-soundcloudResolve :: ByteString -> Proxy Response
-soundcloudResolve url = do
-  e <- view secrets
-  s <- liftIO $ Config.require e "secrets.soundcloud"
-  let req = Req.queryItem "client_id" s . Req.queryItem "url" url $ baseReq
-  mgr <- view manager
-  res <- liftIO $ recovering x2 [handler] $ const (Client.httpLbs req mgr)
+-- get "/proxy/soundcloud/resolve" (continue soundcloudResolve) (query "url")
```
Comment (battermann) [inline services/proxy/src/Proxy/API/Public.hs:136]:
```suggestion
```

**Review by battermann — COMMENTED** [review]
approve!

**Review by fisx — APPROVED** [review]

**Comment by fisx** [comment]
```
https://staging-nginz-https.zinfra.io/v2/proxy/giphy/v1/gifs/search?limit=100&offset=0&q=kitty

403

{
  "data": [],
  "meta": {
    "status": 403,
    "msg": "BANNED",
    "response_id": ""
  }
}
```

i suppose i'll wait for tomorrow?

**Comment by fisx** [comment]
i ran proxy-test.sh on prod: 

[proxy-test.zip](https://github.com/wireapp/wire-server/files/10104617/proxy-test.zip)

ready to merge!  then we can try it "again" on staging with the new proxy.

**Review by battermann — COMMENTED** [review]
LGTM

**Pull request merged**